packages/sign_in_with_apple/macos/Classes/SignInWithApplePlugin.swift

```diff
@@ -100,7 +100,7 @@ public class SignInWithApplePlugin: NSObject, FlutterPlugin {
         } else {
             result(
```

Comment:
For the macOS plugin I thought about reusing the same source code file, potentially with some conditional compilation if needed

Reply:
I was thinking the same. The biggest issue I found is that you can't include external files in a `Podspec`.

Maybe we just need a separate `Podspec` which is also `Flutter` independent?

What I was also thinking is about having two classes for the Plugin. One that is used when signing in with apple is not available, and one that is used when it's possible.

That would improve the usages of our `#available` as those are spread around.

Reply:
I found the following docs on the sharing code issue:
https://github.com/flutter/flutter/issues/52285
https://github.com/flutter/plugins/pull/2618/files?file-filters%5B%5D=.dart&file-filters%5B%5D=.entitlements&file-filters%5B%5D=.h&file-filters%5B%5D=.m&file-filters%5B%5D=.md&file-filters%5B%5D=.plist&file-filters%5B%5D=.png&file-filters%5B%5D=.podspec&file-filters%5B%5D=.swift&file-filters%5B%5D=.xcconfig&file-filters%5B%5D=.xib&file-filters%5B%5D=.yaml&file-filters%5B%5D=dotfile

Reply:
Nice. Also the iOS version still has the bridging header while macOS does not. Don't recall if that was required, or just how the template was generated (as there was quite some back and forth when trying to get the initial implementation building).

Reply:
I will look into this in a future PR